File: dart/difference-of-squares/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {
  num squareOfSum(int n) {
    var sum = 0;
    for (var i = 1; i <= n; i++) {
      sum += i;
    }
    return pow(sum, 2);
  }

  num sumOfSquares(int n) {
    var sum = 0;
    for (var i = 1; i <= n; i++) {
      sum += pow(i, 2).toInt();
    }
    return sum;
  }

  num differenceOfSquares(int i) => squareOfSum(i) - sumOfSquares(i);
}

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
import kotlin.math.pow

class Squares(private val n: Int) {
    fun sumOfSquares(): Int = (1..n).sumOf { it.pow() }

    fun squareOfSum(): Int = (1..n).sum().pow(2)

    fun difference(): Int = squareOfSum() - sumOfSquares()
}

private fun Int.pow(int: Int = 2) : Int = toDouble().pow(int).toInt()
